src/main/java/com/aws/greengrass/mqtt/bridge/clients/LocalMqttClientFactory.java

```diff
@@ -36,21 +43,22 @@ public LocalMqttClientFactory(MQTTClientKeyStore mqttClientKeyStore,
      * @throws MessageClientException if unable to create client
      */
     public MessageClient<MqttMessage> createLocalMqttClient() throws MessageClientException {
-        checkConfig();
+        BridgeConfig config = this.config.get();
+        if (config == null) {
+            throw new IllegalStateException("Bridge configuration not set");
```

Comment:
Illegal state shouldn't be used for this, it has a very very specific meaning.

We always prefer checked exceptions.

Reply:
Can it ever be null?

Reply:
i thought this was the specific meaning 😅   replaced w/ MessageClientException

Reply:
Nevermind that part, I removed it. I was thinking about a different exception type.

Reply:
technically it can't be null, because we set the config during `install` and this factory is called within `startup`. more of a guard in case refactoring happens in the future